src/fastfeedparser/main.py

```diff
@@ -44,640 +47,573 @@ def parse(source):
-                value = get_element_value(item, atom_field, namespaces) or (
-                    (
-                        get_element_value(
-                            item, atom_field, namespaces, attribute="href"
-                        )
-                        or get_element_value(
-                            item, atom_field, namespaces, attribute="link"
-                        )
-                    )
-                    if is_attr
-                    else ""
-                )
-            else:  # RDF
-                value = (
-                    get_element_value(item, rdf_field, namespaces) if rdf_field else ""
-                )
-            return value if value else None
-
-        fields = [
-            (
-                "title",
-                "title",
-                "{http://www.w3.org/2005/Atom}title",
-                "{http://purl.org/rss/1.0/}title",
-            ),
-            (
-                "link",
-                "link",
-                "{http://www.w3.org/2005/Atom}link",
-                "{http://purl.org/rss/1.0/}link",
-                True,
-            ),
-            (
-                "description",
-                "description",
-                "{http://www.w3.org/2005/Atom}summary",
-                "{http://purl.org/rss/1.0/}description",
-            ),
-            (
-                "published",
-                "pubDate",
-                "{http://www.w3.org/2005/Atom}published",
-                "{http://purl.org/dc/elements/1.1/}date",
-            ),
-            (
-                "updated",
-                "lastBuildDate",
-                "{http://www.w3.org/2005/Atom}updated",
-                "{http://purl.org/dc/terms/}modified",
-            ),
+    # Check for guid that looks like a URL
+    guid = item.find("guid")
+    guid_text = guid.text.strip() if guid is not None and guid.text else None
+    is_guid_url = guid_text and guid_text.startswith(("http://", "https://"))
+
+    if is_guid_url:
+        # Prefer guid as link when it looks like a URL
+        entry["link"] = guid_text
+        if alternate_link:
+            entry_links.insert(
+                0, {"rel": "alternate", "type": "text/html", "href": guid_text}
+            )
+    elif alternate_link:
+        entry["link"] = alternate_link["href"]
+        entry_links.insert(0, alternate_link)
+    elif "link" not in entry and guid and guid.get("isPermaLink") == "true":
+        entry["link"] = guid_text
+
+    content = None
+    if feed_type == "rss":
+        content = item.find("{http://purl.org/rss/1.0/modules/content/}encoded")
+        if content is None:
+            content = item.find("content")
+    elif feed_type == "atom":
+        content = item.find("{http://www.w3.org/2005/Atom}content")
+
+    if content is not None:
+        content_type = content.get("type", "text/html")  # Default to text/html
+        if content_type in {"xhtml", "application/xhtml+xml"}:
```

Comment:
BREAKING?: previously checked for "html" "xhtml" but comment below states that we want XHTML, so i replaced html with xhtml mime.